routers/web/repo/issue_view.go

```diff
@@ -397,24 +374,133 @@ func ViewIssue(ctx *context.Context) {
 		}
 	}
 
-	var (
-		role                 issues_model.RoleDescriptor
-		ok                   bool
-		marked               = make(map[int64]issues_model.RoleDescriptor)
-		comment              *issues_model.Comment
-		participants         = make([]*user_model.User, 1, 10)
-		latestCloseCommentID int64
-	)
+	if prepareIssueViewCommentsAndSidebarParticipants(ctx, issue) {
+		return
+	}
```

Comment:
That's really bad naming.

`if preparation succeeds, then return`?

Reply:
The return value means `written`, not success.

Reply:
Readers do not think so, do not bring more surprises to readers. I believe all people with common sense read  `if do something { return }`  as `if something succeeds, then return`

If you are not able to name it clearly, please just check `if ctx.Written()`

Reply:
And there is one more problem, due to the problematic web framework design, if developers forget to write response and only "return true/false" from the "prepare" functions, then end users only sees an 200 empty page.

The only right method is to always check "ctx.Written()", avoid abusing "written" and "return"